models/users.js

```diff
@@ -0,0 +1,16 @@
+import mongoose from "mongoose";
+const { Schema } = mongoose;
+
+const Users = new Schema(
+    {
+        username: { type: String, unique: true, required: true, trim: true },
+        firstname: { type: String, required: true },
+        lastname: { type: String, required: true },
+        emailAddress: { type: String, unique: true, required: true },
+        password: { type: String, unique: true, required: true, trim: true },
+        role: { type: String, required: true }
```

Comment:
`role` here should be an `enum` type. i.e `role: {type: String, enum: ['admin', 'regular'], required: true}`